ReactiveCocoa/NSObject+KeyValueObserving.swift

```diff
@@ -55,6 +55,113 @@ extension Reactive where Base: NSObject {
 			return disposable
 		}
 	}
+
+	#if swift(>=3.2)
+	private func producer<U>(
+		for keyPath: KeyPath<Base, U>,
+		action: @escaping (Signal<U, NoError>.Observer) -> (Any?) -> Void
```

Comment:
Since we're always calling `observer.send(value: $0 as X)`, could we change this to be more like a `map` instead? Transform the value (`(Any?) -> U`) and then send it on the observer?